dts/bindings/misc/zephyr,flash-disk.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: Block storage device on flash partition
+
+compatible: "zephyr,flash-disk"
+
+include: base.yaml
+
+properties:
+    partition:
+      type: phandle
+      required: true
+      description: |
+        Backing storage flash map partition.
+
+    disk-name:
+      type: string
+      required: true
+      description: |
+        Disk name.
+
+    block-size:
```

Comment:
This property really should be called sector-size. The flash_disk implementation returns this value when it is requested to return `DISK_IOCTL_GET_SECTOR_SIZE`. The disk api has nothing that is called block-size, introducing this is a bad idea for this PR. If a rename/api change is required this should be done in a separate PR.

Reply:
@jfischer-no Should I change it so it is consistent with the naming used in other part of the code (i.e. "consistently wrong")?

Reply:
@desowin I leave it up to you whether `block-size` or `sector-size`. AFAIK we cannot introduce new binding property and then rename. It should be clear what it is about and ´sector-size´ is ambiguous and can be confused with property of a flash device here. This driver emulates a block device and that is why I think it should not be called `sector-size`.

Reply:
This driver does not emulate a block device, it emulates a disk. The `sector-size` is not ambiguous in the case of a disk.